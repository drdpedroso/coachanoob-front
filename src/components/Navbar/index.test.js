import React from 'react';
import renderer from 'react-test-renderer';
import ResponsiveContainer from './index'
import {render, fireEvent} from '@testing-library/react'
import {MemoryRouter} from "react-router-dom";


describe('NavBar', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<MemoryRouter initialEntries={['/']}><ResponsiveContainer>
                <div> Hey!</div>
            </ResponsiveContainer></MemoryRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should change active route when clicking on the menu item', () => {
        const component = render(<MemoryRouter
            initialEntries={['/']}><ResponsiveContainer> </ResponsiveContainer></MemoryRouter>)
        const homeLink = component.getByTestId('link-to-home')
        const searchLink = component.getByTestId('link-to-search')
        expect(homeLink.className.includes('active')).toBe(true)
        fireEvent.click(searchLink)
        expect(homeLink.className.includes('active')).toBe(false)
        expect(searchLink.className.includes('active')).toBe(true)
    })
})
