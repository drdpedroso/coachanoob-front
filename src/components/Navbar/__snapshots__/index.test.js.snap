// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders correctly 1`] = `
<div>
  <div>
    <div>
      <div
        className="ui inverted vertical center aligned segment sc-bdVaJa fGrPIJ"
      >
        <div
          className="ui large inverted pointing secondary menu"
        >
          <div
            className="ui container"
          >
            <a
              className="active item"
              data-testid="link-to-home"
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
            <a
              className="item"
              data-testid="link-to-search"
              href="/user"
              onClick={[Function]}
            >
              Pesquisar anúncios
            </a>
            <a
              className="item"
              data-testid="link-to-my-listing"
              href=""
              onClick={[Function]}
            >
              Meus anúncios
            </a>
            <a
              className="item"
              data-testid="link-to-my-proposal"
              href=""
              onClick={[Function]}
            >
              Minhas Propostas
            </a>
            <a
              className="item"
              data-testid="link-to-how-it-works"
              href=""
              onClick={[Function]}
            >
              Como Funciona?
            </a>
            <div
              className="right item"
              onClick={[Function]}
            >
              <a
                className="ui inverted button"
                data-testid="logout-button"
                onClick={[Function]}
                role="button"
                style={
                  Object {
                    "marginLeft": "0.5em",
                  }
                }
              >
                Sair
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div>
       Hey!
    </div>
  </div>
</div>
`;
